### PR TITLE
fix(2625): Refactor webhook test

### PR DESCRIPTION
**Pull request by itleigns**
## Context
Many tests about [webhooks/helper.js](https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/webhooks/helper.js) are in [webhooks.test.js](https://github.com/screwdriver-cd/screwdriver/blob/master/test/plugins/webhooks.test.js). These need to be moved to [webhooks.helper.test.js](https://github.com/screwdriver-cd/screwdriver/blob/master/test/plugins/webhooks.helper.test.js).

## Objective
- Tests about `webhooks/helper.js` in `webhooks.test.js` are moved to `webhooks.helper.test.js`.
- We mock functions of `webhooks/helper.js` in `webhooks.test.js` and `processHooks.test.js`.
- We delete unnecessary json.
  - These json are only used as input to [scm.parseHook](https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/webhooks/index.js#L72).

## References
#2625

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: test/plugins/processHooks.test.js [hunk 1]
```diff
@@ -69,76 +30,22 @@ describe('processHooks plugin test', () => {
...
-            branch: Promise.resolve('master')
-        });
+        userFactoryMock = sinon.stub();
+        eventFactoryMock = sinon.stub();
+
+        startHookEventMock = sinon.stub();
 
         plugin = rewire('../../plugins/processHooks');
 
+        plugin.__set__('startHookEvent', startHookEventMock);
```
Comment (itleigns) [inline test/plugins/processHooks.test.js:47]: `startHookEvent` is mocked here.

File: test/plugins/webhooks.test.js [hunk 2]
```diff
@@ -163,2314 +134,109 @@ describe('webhooks plugin test', () => {
...
-
-            it('returns 204 when getCommitRefSha() is rejected', () => {
-                pipelineFactoryMock.scm.getCommitRefSha.rejects(new Error('some error'));
-
-                return server.inject(options).then(reply => {
-                    assert.equal(reply.statusCode, 204);
-                    assert.notCalled(eventFactoryMock.create);
-                });
+            return server.inject(options).then(() => {
+                assert.calledOnce(pipelineFactoryMock.scm.parseHook);
```
Comment (yk634) [inline test/plugins/webhooks.test.js]: Is this assertion necessary?

Comment (itleigns) [reply test/plugins/webhooks.test.js]: There are too many test cases that check if parsedhook is called, so I will remove some of them. However, I want to check it somewhere, so I will keep one of them.

**Review by yk634 — DISMISSED** [review]

**Comment by itleigns** [comment]
I modified the test to correspond to #2641.

**Review by yk634 — APPROVED** [review]

**Review by ibu1224 — APPROVED** [review]
LGTM

**Review by wahapo — APPROVED** [review]

**Review by yoshwata — APPROVED** [review]
lgtm

**Review by tkyi — APPROVED** [review]

**Pull request merged**